trade-finance-manager-api/src/v1/controllers/update-issued-facilities.js

```diff
@@ -11,115 +11,97 @@ const updatedIssuedFacilities = async (deal) => {
-        const facilityStatusUpdate = CONSTANTS.FACILITIES.FACILITY_STATUS_PORTAL.ACKNOWLEDGED;
-        const updatedFacilityStatus = await api.updatePortalFacilityStatus(facilityId, facilityStatusUpdate);
-
-        // add acknowledged flag, update Portal facility
-        const portalFacilityUpdate = {
-          hasBeenAcknowledged: true,
-        };
-
-        const updatedPortalFacility = await api.updatePortalFacility(facilityId, portalFacilityUpdate);
-
-        // update TFM facility
-        const facilityExposurePeriod = await getFacilityExposurePeriod(facility);
-
-        const facilityGuaranteeDates = getGuaranteeDates(facility, dealSubmissionDate);
-
-        const facilityPremiumSchedule = await getFacilityPremiumSchedule(
-          facility,
-          facilityExposurePeriod,
-          facilityGuaranteeDates,
-        );
-
-        const facilityUpdate = {
-          ...portalFacilityUpdate,
-          ...facilityExposurePeriod,
-          facilityGuaranteeDates,
-          premiumSchedule: facilityPremiumSchedule,
-        };
-
-        await api.updateFacility(facilityId, facilityUpdate);
-
-        // update object to return in response
-        const updatedFacilityResponseObj = {
-          ...updatedFacilityStatus,
-          ...updatedPortalFacility,
-          tfm: {
-            ...facilityUpdate,
-          },
-        };
-
-        if (facilityType === 'bond') {
-          bonds.push(updatedFacilityResponseObj);
-        } else if (facilityType === 'loan') {
-          loans.push(updatedFacilityResponseObj);
-        }
-
-        allUpdatedFacilities.push(updatedFacilityResponseObj);
-        updatedCount += 1;
-      } else if (facilityType === 'bond') {
-        // update deal object to return in response
-        bonds.push(facility);
-      } else if (facilityType === 'loan') {
-        loans.push(facility);
-      }
-
-      if (shouldUpdateCount === updatedCount) {
-        modifiedDeal.dealSnapshot.bondTransactions.items = bonds;
-        modifiedDeal.dealSnapshot.loanTransactions.items = loans;
-
-        await sendIssuedFacilitiesReceivedEmail(
-          modifiedDeal,
-          allUpdatedFacilities,
-        );
-
-        return resolve(modifiedDeal);
-      }
-      return modifiedDeal;
-    });
-  });
+  } = deal;
+
+  const updatedFacilities = [];
+
+  modifiedDeal.facilities = await Promise.all(modifiedDeal.facilities.map(async (facility) => {
+    let modifiedFacility = facility;
+
+    const {
+      _id: facilityId,
+      facilityStage,
+      previousFacilityStage,
+      hasBeenAcknowledged,
+    } = modifiedFacility;
+
+    // we only need to update issued facilities if:
+    // - a facility has changed from `Unissued` to `Issued` (`Conditional` to `Unconditional` for a loan)
+    // - and the facility changing from `Unissued` to `Issued` has NOT been previously acknowledged
+
+    const bondIsIssued = ((previousFacilityStage === CONSTANTS.FACILITIES.FACILITY_STAGE_PORTAL.UNISSUED)
```

Comment:
There's a helper for determining if a facility is issued
https://github.com/foundry4/dtfs2/blob/master/trade-finance-manager-api/src/v1/helpers/issued-facilities.js

Reply:
Could even add to the facility mappings `isIssued` - GEF facilities already have this. I'll update this function to use the `issued-facilities` function for now - next PR will be for facilities

Reply:
The logic in update-issued-facilities only applies to facilities that have been previously unissued and then changed to issued, so existing helper would return false positives. Created a new helper: 'was-previously-unissued' and used with 'is-issued', a lot cleaner 👍